### PR TITLE
ENH add multiple variants in key_add

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Added a ``news`` entry

This PR does two things. First, it allows key_add migrators to add more than one key. Second, it fixes a bug where smithy is applying the recipe's CBC on top of the global pinning file + recipe CBC + migration. Sometimes this causes the changes from the migration to be removed when rerendering happens.

### CODE REVIEW

File: conda_smithy/configure_feedstock.py [hunk 1]
```diff
@@ -371,8 +370,8 @@ def _collapse_subpackage_variants(
     _trim_unused_zip_keys(used_key_values)
     _trim_unused_pin_run_as_build(used_key_values)
 
-    logger.debug("top_level_loop_vars {}".format(top_level_loop_vars))
-    logger.debug("used_key_values {}".format(used_key_values))
+    logger.info("top_level_loop_vars {}".format(top_level_loop_vars))
+    logger.info("used_key_values {}".format(used_key_values))
```
Comment (beckermr) [inline conda_smithy/configure_feedstock.py:374]:
```suggestion
    logger.debug("top_level_loop_vars {}".format(top_level_loop_vars))
    logger.debug("used_key_values {}".format(used_key_values))
```

**Comment by beckermr** [comment]
OK @isuruf @leofang This PR will allow us to do two things.

1. We can convert the cuda 11.1 and 11.2 migrator to a key_add migration, which it probably should have been in the first place.
2. We can now support variants specified in the recipe's CBC and the migrator in smithy so that we can render cudnn properly. See this PR (https://github.com/conda-forge/cudnn-feedstock/pull/14) for what that looks like. 

Ready for review!

File: conda_smithy/configure_feedstock.py [hunk 2]
```diff
@@ -615,17 +614,36 @@ def _render_ci_provider(
...
-            permit_undefined_jinja=True,
-            finalize=False,
-            bypass_env_check=True,
-            channel_urls=forge_config.get("channels", {}).get("sources", []),
-        )
+        # AFAIK there is no way to get conda build to ignore the CBC yaml
+        # in the recipe. This one can mess up migrators applied with local
+        # CBC yaml files where variants in the migrators are not in the CBC.
+        # Thus we move it out of the way.
+        # TODO: upstream this as a flag in conda-build
```
Comment (isuruf) [inline conda_smithy/configure_feedstock.py:621]: This is just wrong. We need to use the CBC yaml to render the recipe.

Comment (beckermr) [reply conda_smithy/configure_feedstock.py:621]: That is done above here AFAIK: https://github.com/conda-forge/conda-smithy/pull/1454/files#diff-dc66a1980eec62725cfb8b6faea5b3c90dc7d1009d1229f5f55eecfb267a6677R607

Comment (beckermr) [reply conda_smithy/configure_feedstock.py:621]: The CBC will remove variants added by migrators if the keys overlap.

Comment (beckermr) [reply conda_smithy/configure_feedstock.py:621]: See this line in conda build: https://github.com/conda/conda-build/blob/master/conda_build/variants.py#L213

Comment (beckermr) [reply conda_smithy/configure_feedstock.py:621]: I've printed out all of the config variants in each step and without this block, we definitely cannot support rendering migrators with extra keys on top of local CBCs. That is a fine conclusion, but it is the CBC that is removing extra variants for cudnn.

Comment (isuruf) [reply conda_smithy/configure_feedstock.py:621]: Makes sense.

**Comment by isuruf** [comment]
Thanks

**Comment by leofang** [comment]
Thanks, Matt! I would never figure this out in full by myself...

**Pull request merged**